hnn_core/network.py

```diff
@@ -819,8 +822,12 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
         Attached drive is stored in self.external_drives[name]
         self.pos_dict is updated, and self._update_gid_ranges() called
         """
+        # clear connectivity for this drive if it already exists
         if name in self.external_drives:
-            raise ValueError(f"Drive {name} already defined")
```

Comment:
why did you change this behavior? Was it necessary so that the optimization engine could reset drives? Does this also make sense for someone using just `simulate_dipole` or do we need some safety mechanism there?

Reply:
My justification here is up for debate. The whole idea with passing an instance of `Network` into `optimize_evoked()` is that optimization will modify and return a `Network` instance with the exact same drives, except with optimal timing and weight parameters. All other aspects, including the name of each drive, remain untouched. While this could be accomplished by first retrieving the drive parameters, clearing all drives from the original `Network` object, and then re-adding the modified drives, I felt that this was an unnecessary complication in the API for modifying a drive.

Generally speaking, I think users should be able to modify a drive without having to remove a specific drive and then re-add a modified version. As is the case with GIDs or file names, the whole point of having a drive name is that it is a unique identifier that, should the use repetitively, creates a new drive instance or modifies a previously existing one with the same name. Maybe this is a bigger discussion that should wait for its own PR....

Reply:
Any thoughts @jasmainak or @ntolley? I'm happy to revert this if either of you'd prefer.

Reply:
will this update the drive with default values for the parameters that are not passed? For an update operation, I would think it's more intuitive to retain the old values for every parameter except the one updated. I'm thinking you could have:

```py
net.update_evoked_drive(tstart_std=True)
```

and that does the update. Alternative is to have a `force_update=True` parameter. I'm drawing a bit of inspiration from database APIs where there is a distinction between insert and update operations.

Reply:
Yes, it currently overwrites the drive completely with whatever default behavior is contained in the `add_evoked_drive` method.

IDK that we really want to make this PR any bigger, so if we want to create a separate method for updating drive params, my vote would be to revert my previous changes for now and address this later.

Reply:
sounds good, we can raise an issue and perhaps it can be discussed when we discuss the drives API !

Reply:
Reverted/fixed in https://github.com/jonescompneurolab/hnn-core/pull/446/commits/e85af4876a9754ed7370eb2e86f22305c8322fc8, https://github.com/jonescompneurolab/hnn-core/pull/446/commits/1f4769203047350e750671c343d539d5148bce7b, and https://github.com/jonescompneurolab/hnn-core/pull/446/commits/daecdcba8d751afce68526770a010918d1021687.

Reply:
Let's discuss more later. I think the only time this has become bothersome is when working in jupyter notebooks. If you define you drives in a separate cell then you always need to rerun the cell that instantiates the network model.